backend/src/apiserver/client/argo.go

```diff
@@ -18,40 +18,42 @@ import (
 	"time"
 
 	argoclient "github.com/argoproj/argo/pkg/client/clientset/versioned"
-	"github.com/argoproj/argo/pkg/client/clientset/versioned/typed/workflow/v1alpha1"
+	argoprojv1alpha1 "github.com/argoproj/argo/pkg/client/clientset/versioned/typed/workflow/v1alpha1"
 	"github.com/cenkalti/backoff"
 	"github.com/golang/glog"
 	"github.com/pkg/errors"
 	"k8s.io/client-go/rest"
 )
 
-func CreateWorkflowClient(namespace string) (v1alpha1.WorkflowInterface, error) {
-	restConfig, err := rest.InClusterConfig()
-	if err != nil {
-		return nil, errors.Wrap(err, "Failed to initialize workflow client.")
-	}
-	wfClientSet := argoclient.NewForConfigOrDie(restConfig)
-	wfClient := wfClientSet.ArgoprojV1alpha1().Workflows(namespace)
-	return wfClient, nil
+type ArgoClientInterface interface {
+	Workflow(namespace string) argoprojv1alpha1.WorkflowInterface
+}
+
+type ArgoClient struct {
+	argoProjClient argoprojv1alpha1.ArgoprojV1alpha1Interface
 }
 
-// creates a new client for the Kubernetes Workflow CRD.
-func CreateWorkflowClientOrFatal(namespace string, initConnectionTimeout time.Duration) v1alpha1.WorkflowInterface {
-	var wfClient v1alpha1.WorkflowInterface
-	var err error
+func (argoClient *ArgoClient) Workflow(namespace string) argoprojv1alpha1.WorkflowInterface {
+	return argoClient.argoProjClient.Workflows(namespace)
+}
+
+func NewArgoClient(initConnectionTimeout time.Duration) *ArgoClient {
```

Comment:
NewArgoClient -> NewArgoClientOrFatal
Since fatal will lead to server crash.

Reply:
Done.